File: JS-MongoDB/updateAndReaplceGenerator.js
var __dataframe = require("dataframe-js").DataFrame;
var __nosql = require("mongodb");
var __fs = require("fs")
var __parse = require("csv-parse");

const func = async () => {

	const __nosqlClient = new __nosql.MongoClient(
		"mongodb://127.0.0.1:27017/",
		{ useUnifiedTopology: true }
	);
	
	await __nosqlClient.connect();
	
	const nosql = __nosqlClient.db("mydb").collection("weather");

    var statement = "[ " + "{ \"name\": \"Antonio\", \"surname\": \"Cirillo\", \"age\": 21 }" + ", " + "{  \"name\": \"Giovanni\", \"surname\": \"Rapa\", \"age\": 21 }" + " ]"
	let insertStudents;
	if(statement.charAt(0) === "[")
		insertStudents = JSON.parse(statement);
	else
		insertStudents = JSON.parse("[" + statement + "]");
	
	await nosql.insertMany(insertStudents); 

    var updateFilter = "{ \"name\": \"Antonio\" }"
	var updateStatement = "{ \"$set\": { \"name\": \"Marcello\" } }"
	const updateQueryFilter = JSON.parse(updateFilter);
	
	const updateQuery = JSON.parse(updateStatement);
	
	(await nosql.updateMany(updateQueryFilter, updateQuery));
	
	var replaceFilter = "{ \"name\": \"Marcello\" }"
	var replaceStatement = "{ \"name\": \"Antonio\", \"surname\": \"Montella\" }"
	const replaceQueryFilter = JSON.parse(replaceFilter);
	
	const replaceQuery = JSON.parse(replaceStatement);
	
	(await nosql.replaceOne(replaceQueryFilter, replaceQuery));
	
    await __nosqlClient.close(); 

}

func(); 

File: PY-MongoDB/updateAndReaplceGenerator.py
#import os
#import random
#import time
#import math 
import pandas as pd
#import numpy as np
import json
#import urllib.request
#import pymysql
from pymongo import MongoClient

#import azure.functions as func
#from azure.storage.queue import QueueServiceClient

def main() :
	#__environment = 'azure'
	#__queue_service = QueueServiceClient(account_url='https://"${storageName}".queue.core.windows.net', credential='"${storageKey}"')
	#__event = req.get_json()
	#data = __event['data']
	nosqlClient = MongoClient('mongodb://127.0.0.1:27017/') 
	nosqlDatabase = nosqlClient['mydb']
	nosql = nosqlDatabase['weather']
	updateFilter =  '{ \"name\": \"Antonio\" }' 
	updateStatement =  '{ \"$set\": { \"name\": \"Marcello\" } }' 
	updateQueryFilter = json.loads(updateFilter)
	
	updateQuery = json.loads(updateStatement)
	
	nosql.update_many(updateQueryFilter, updateQuery)
	
	replaceFilter =  '{ \"name\": \"Marcello\" }' 
	replaceStatement =  '{ \"name\": \"Antonio\", \"surname\": \"Montella\" }' 
	replaceQueryFilter = json.loads(replaceFilter)
	
	replaceQuery = json.loads(replaceStatement)
	
	nosql.replace_one(replaceQueryFilter, replaceQuery)

	#__queue_service_client = __queue_service.get_queue_client('termination-"${function}"-"${id}"')
	#__queue_service_client.send_message('terminate')
main()